### ISSUE TITLE
Bug in Consolidator

### ISSUE THREAD

**Issue by Swader** [Swader/rmrk-tools#19]
When the mint and mintnft op are in the same block, the consolidator tends to forget one and think the nft doesn't have a parent collection. This has resulted in a bunch of false negatives.

**Comment by Swader** [comment]
This is not a bug in the consolidator, but in the fetcher.

**Comment by Swader** [comment]
This is not a bug in the fetcher after all, Kodadot have been using batch instead of batchall. Considering making an exception since it's a major mess with a lot of invalid NFTs.

**Comment by Swader** [comment]
The fetched *does* have a bug too: in looking for "extras" alongside a remark in a batch, it will override the main remark TX if it finds another.

Batch 1:

- remark
- non-remark action 1
- non-remark action 2

Batch 2:

- remark 1
- remark 2
- non-remark action 1
- non-remark action 2

Batch 1 will be processed as:

remark.extras = [non-remark action 1, non-remark action 2]

Batch 2 will be processed as:

remark 2.extras = [non-remark action 1, non-remark action 2]

In other words, the first remark is overwritten by the second.

Needs workaround.

**Comment by Swader** [comment]
Proposed solution:

- detect multiple remarks in a batch
- if detected, push all but the last one into the blockcalls array
- only process extras for last one - consider all extras applied to last one

Applying fix...

**Comment by Swader** [comment]
Fixed in #22
